channel/allocation.go

```diff
@@ -123,7 +122,7 @@ func NewAllocation(numParts int, assets ...Asset) *Allocation {
 // AssetIndex returns the index of the asset in the allocation.
 func (a *Allocation) AssetIndex(asset Asset) (Index, bool) {
 	for idx, _asset := range a.Assets {
-		if ok, err := equalMarshaling(asset, _asset); ok && err == nil {
+		if ok, err := perunio.EqualBinary(asset, _asset); ok && err == nil {
```

Comment:
We are currently working on a #279  that introduces the Asset type with an Equal method. This should be then used here, depending on which we merge first.

Reply:
Yes, we could replace this instance with `Asset.Equal` method.

However, regarding which PR to merge first, I would suggest we do this one before #279. Because, though we don't require `equalMarshalling` function for comparing two assets, we still need it for comparing two data. See https://github.com/hyperledger-labs/go-perun/pull/272#discussion_r763697134